spark/src/main/scala/geotrellis/spark/util/TaskUtils.scala

```diff
@@ -17,32 +17,31 @@
 package geotrellis.spark.util
 
 
-import java.util.concurrent.Executors
+import cats.effect._
+import cats.syntax.all._
+
 import scala.concurrent.duration._
-import scalaz._
-import scalaz.concurrent._
-import scalaz.stream._
-import scalaz.stream.async._
-import collection.JavaConversions._
 import scala.util.Random
 
 object TaskUtils extends App {
-   implicit class TaskBackoff[A](task: Task[A]) {
-    /**
-     * Implement non-blocking Exponential Backoff on a Task.
-     * @param  p  returns true for exceptions that trigger a backoff and retry
-     */
-    def retryEBO(p: (Throwable => Boolean) ): Task[A] = {
-      def help(count: Int): Future[Throwable \/ A] = {
+  /**
+    * Implement non-blocking Exponential Backoff on a Task.
+    * @param  p  returns true for exceptions that trigger a backoff and retry
+    */
+
+  implicit class IOBackoff[A](ioa: IO[A]) {
+    def retryEBO(p: (Throwable => Boolean))(implicit timer: Timer[IO]): IO[A] = {
```

Comment:
NOTE: Double check how it works

Reply:
Has this comment been rectified?  Did you double-check?

Reply:
Yea, looks good, and here is the reference: https://typelevel.org/cats-effect/datatypes/io.html#example-retrying-with-exponential-backoff